### DISCUSSION TITLE
Fixing airflow docker image vulnerabilities detected by trivy scanner

### DISCUSSION THREAD

**Discussion by vineethguna** [apache/airflow#15891] [Q&A]
Hi,

I have found vulnerabilities while scanning the airflow docker images using [trivy](https://github.com/aquasecurity/trivy) scanner

Airflow versions on which vulnerabilities are found - 1.10.x, 2.0.x

Airflow images are pulled from [docker hub](https://hub.docker.com/r/apache/airflow)

Any suggestions on how to fix these vulnerabilities?

**Comment by potiuk** [comment]
If you checked that those are real vulnerabilties that import production running of airflow, please report them using the process described here: https://www.apache.org/security/ - one vulnerability per report please. We have an established security vulnerability process handling in the foundation. 

Note that the ASF security team will reject "bulk" reports and security scan results usually because the security scanners are very prone to false positives. You should check if the security vulnerability reported by your tool are actual vulnerabilities and report the "real" ones one-by-one using the process described there.

**Reply by vineethguna** [reply]
Thanks @potiuk for the guidance

The vulnerabilities which the scanner reported are against the system packages used in docker image (for example openssl, libxml2 etc)

So if we want to make sure we use a secure image of airflow can we upgrade these system packages and create a new image out of the base airflow image?

Something like this
```
FROM apache/airflow:1.10.12-python3.6
USER root
RUN apt-get update
RUN apt-get install -y --only-upgrade bind9-host=1:9.11.5.P4+dfsg-5.1+deb10u5
RUN apt-get install -y --only-upgrade apt=1.8.2.2
```

Only concern which I see is if upgrading any system package would effect the functionality of airflow

**Reply by potiuk** [reply]
The only recommended approach I can suggest is to migrate to `1.10.15` version - which is newer, based on more recent "python" images which contains more recent security patches. 

Upgrading dependencies on your own makes you responsible for any problems you have - you are basically on your own.

However I strongly recommend to migrate first to 1.10.15 and then immediately to 2.1 version (which we are about to release - it will be available  in ~ 1 week). As you can read here: https://airflow.apache.org/docs/apache-airflow/stable/upgrading-to-2.html#step-2-upgrade-to-airflow-1-10-15-a-k-a-our-bridge-release Airflow 1.10.* line is supported currently only for critical security fixes till 24th June 2021. As of June 2021 the only supported version of Airflow will be 2.*

**Reply by vineethguna** [reply]
Thanks @potiuk this helps
